### DISCUSSION TITLE
Windows Build failed with bindgen issue

### DISCUSSION THREAD

**Discussion by LokeshNagireddy** [actions/virtual-environments#10095] [General questions]
### Description

Hi Team,

My build failed with below error even though im using the latest image.

![image](https://github.com/actions/runner-images/assets/113025693/9496099d-7e19-45d4-aeec-ed2e0f35e163)

image ref: 870d08d9cbff9a9a2b0e805c7e41e5238c184bf0

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] Ubuntu 24.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

Windows server 2022

### Is it regression?

870d08d9cbff9a9a2b0e805c7e41e5238c184bf0

### Expected behavior

Successful build without errors

### Actual behavior

Error with bindgen error

### Repro steps

Just run the build with version 870d08d9cbff9a9a2b0e805c7e41e5238c184bf0

**Comment by kishorekumar-anchala** [comment]
Hi Lokesh , 

  we are unable to produce any error with given [commit](https://github.com/actions/runner-images/commit/870d08d9cbff9a9a2b0e805c7e41e5238c184bf0), could you please take the latest code from main branch.

According to your error , it showing that there is a space in the path or spelling, please look into it .

**Comment by LokeshNagireddy** [comment]
Hi @kishorekumar-anchala, 

The build failed while performing Runall-tests after doing a windows update and  as given in windows-2022 packer file.

![image](https://github.com/actions/runner-images/assets/113025693/3c3f5160-f7f6-4281-8ec7-18b2eb2e22f6)

I'm using the latest code given for windows 2022.

Therefore, the error showing is from one of the tests or some check after the test, and not from our side.

im providing some more log from previous successful step(Wix.tests.ps1) for your reference.

Running tests from 'C:\image\tests\Wix.Tests.ps1'
Describing Wix
  [+] Wix Toolset version from registry 18ms (15ms|3ms)
  [+] Wix variable exists 16ms (15ms|1ms)
  [+] Wix binaries folder exists 18ms (17ms|1ms)
  [+] Wix binaries folder is in PATH 17ms (16ms|1ms)
Tests completed in 468.27s
Tests Passed: 604, Failed: 0, Skipped: 13, Inconclusive: 0, NotRun: 0
Provisioning with Powershell...
Provisioning with powershell script: /tmp/powershell-provisioner3754239793
Provisioning with Powershell...
Provisioning with powershell script: /tmp/powershell-provisioner3591480050
PHP version 8.3.8 (c:\tools\php\php.exe)
Run the "diagnose" command to get more detailed diagnostics output.
Found a space in the home path. We heavily discourage this, due to multiple
    incompatibilities. Please check our docs for more information on this, as
    well as some pointers on how to migrate to a different home path.
[31;1mbindgen : [31;1mThe term 'bindgen' is not recognized as a name of a cmdlet, function, script file, or executable program.[0m
[31;1m[31;1mCheck the spelling of the name, or if a path was included, verify that the path is correct and try again.[0m
[31;1m[31;1mAt C:\image\SoftwareReport\SoftwareReport.Common.psm1:53 char:5[0m
[31;1m[31;1m+     bindgen --version | Get-StringPart -Part 1[0m
[31;1m[31;1m+     ~~~~~~~[0m
[31;1m[31;1m+ CategoryInfo          : ObjectNotFound: (bindgen:String) [], ParentContainsErrorRecordException[0m
[31;1m[31;1m+ FullyQualifiedErrorId : CommandNotFoundException[0m
Provisioning step had errors: Running the cleanup provisioner, if present...

May you please have a deep look and let me know.

i look forward for your response.

Thank you,

**Comment by LokeshNagireddy** [comment]
Hi @kishorekumar-anchala, May you please provide an update. Our deployments are stopped with this error.

**Comment by LokeshNagireddy** [comment]
Hi @erik-bershel , May you please have a look at this issue.. im waiting from last 4 days for a solution of this issue.

i have run the build with the latest reference (89d72011e0cd8e10a4f178a194c04f873bf3757d), but still got some test fail issue with svn.

error details:
  Describing Svn
  [-] svn 95ms (90ms|6ms)
  Command 'svn --version --quiet' has finished with exit code 1 and output:
      'svn' is not recognized as an internal or external command,
        operable program or batch file.
    at "svn --version --quiet" | Should -ReturnZeroExitCode, C:\image\tests\ChocoPackages.Tests.ps1:69
      at <ScriptBlock>, C:\image\tests\ChocoPackages.Tests.ps1:69


i look forward for your response.

**Comment by kishorekumar-anchala** [comment]
Hi @LokeshNagireddy ,

   Pipelines got succeeded from main branch we're unable to see your error , kindly fork latest .

**Comment by LokeshNagireddy** [comment]
Hi @kishorekumar-anchala,

I have taken a new image reference from june 18th (https://github.com/actions/runner-images/commit/89d72011e0cd8e10a4f178a194c04f873bf3757d) from main branch.

We had a new issue with svn as shown below.

error details:
Describing Svn
[-] svn 95ms (90ms|6ms)
Command 'svn --version --quiet' has finished with exit code 1 and output:
'svn' is not recognized as an internal or external command,
operable program or batch file.
at "svn --version --quiet" | Should -ReturnZeroExitCode, C:\image\tests\ChocoPackages.Tests.ps1:69
at , C:\image\tests\ChocoPackages.Tests.ps1:69

if the build was success on your side, May you please provide the pipeline detail, so that i will check and compare them.

i look forward for your response.

**Comment by LokeshNagireddy** [comment]
Hi @kishorekumar-anchala , Gentle reminder.

**Comment by kishorekumar-anchala** [comment]
Hi @LokeshNagireddy ,

    Apologies , We haven't reproduce the error from main branch, we're expecting that you made some change in your branch (from where you're running the pipeline) . Could you please try to reproduce with the latest code(main).

**Comment by erik-bershel** [comment]
Hey @LokeshNagireddy!
Take a look here:
[Windows-2019](https://dev.azure.com/github/runner-images/_build?definitionId=64),
[Windows-2022](https://dev.azure.com/github/runner-images/_build?definitionId=97).
SVN commit: https://github.com/actions/runner-images/commit/2e3a8e1ccfd026ace20f6281152ca0d81a6eafd8

We do not support custom builds. If you make any changes to the code, then it is your responsibility to keep it working. We only make sure that the code from the last state of the `main` branch is functional. Now the problem you described does not exist neither in code, or test-builds. Try to wait for other users - maybe someone has encountered something similar in their fork.

**Reply by LokeshNagireddy** [reply]
Hi @erik-bershel , Thanks for your response.
By the way, i dont expect your support for our custom builds.
However, i didnt got the answer for actual bindgen issue, its happenings in Wix.tests file which is not our custom code.
we are using the code from the image as-is.

We are getting the below error while running the Wix.Tests.ps1.

Our pipeline
![image](https://github.com/actions/runner-images/assets/113025693/759501eb-7192-4653-ad39-034f33d3af6d)

But, when i checked your pipeline, there is no error.
![image](https://github.com/actions/runner-images/assets/113025693/8a1ca437-04c7-4d7f-bb62-a72da035b1e1)

As there are no additional code from my end, im confused at this state.

May you please provide some insight, if you got something.

**Reply by erik-bershel** [reply]
I would go the following way, namely, I would look for answers to the following questions:
- when exactly does the problem actually arise?
- was the `bindgen` actually installed ([example](https://dev.azure.com/github/runner-images/_build/results?buildId=136203&view=logs&j=747417c8-a650-5aed-afed-d9b52dbd7d24&t=f05f5d13-c61c-51fd-2a8c-71d3e56d3e3f&l=4519))? 
- did it pass the initial smoke test ([example](https://dev.azure.com/github/runner-images/_build/results?buildId=136203&view=logs&j=747417c8-a650-5aed-afed-d9b52dbd7d24&t=f05f5d13-c61c-51fd-2a8c-71d3e56d3e3f&l=4533))?
- where did it go after installation and passing the test?

**Reply by LokeshNagireddy** [reply]
Hi @erik-bershel ,

Please find the details below.

1. when exactly does the problem actually arise?
The Problem raised when the pipeline executing the SoftwareReport.Common.psm1 script.
![image](https://github.com/actions/runner-images/assets/113025693/3fe49e0b-7c2d-400c-865d-fd7e7714dade)

2. was the bindgen actually installed?
yes, bindgen was successfully installed.
![image](https://github.com/actions/runner-images/assets/113025693/c7f00094-951f-460d-9941-d321e1493e77)

3. did it pass the initial smoke test?
yes, smoke test was successful.
![image](https://github.com/actions/runner-images/assets/113025693/27d09302-bd1c-4bb8-9ebe-8e66537979f6)

4. where did it go after installation and passing the test?
After running tests, it continued to install sbt.
![image](https://github.com/actions/runner-images/assets/113025693/1d747fc0-0acb-4859-9b63-7adb54c11fb7)

Almost all install steps run similar to your pipeline, only difference is at the software report generation.

Please let me know if you need more info.
